src/dynsight/_internal/analysis/rdf.py

```diff
@@ -0,0 +1,179 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+from ovito.io import import_file
+from ovito.modifiers import (
+    CoordinationAnalysisModifier,
+    DeleteSelectedModifier,
+    ExpressionSelectionModifier,
+    LoadTrajectoryModifier,
+    TimeAveragingModifier,
+)
+from scipy.signal import find_peaks
+
+
+class RDF:
+    """Object for computing and analyzing the Radial Distribution Function.
+
+    The Radial Distribution Function (RDF) describes how particle density
+    varies as a function of the distance.
+
+    Parameters:
+        trajectory (list[Path]):
+            List of file paths including trajectory and optional topology file.
+
+        pipeline (Pipeline):
+            OVITO pipeline used for importing and modifying the
+            trajectory data.
+
+        pair_distances (np.ndarray):
+            Array of distances between particle pairs.
+
+        rdf (np.ndarray):
+            Array of RDF values corresponding to the pair distances.
+
+        minima_points (tuple):
+            Tuple containing arrays of minima point distances and RDF values.
+    """
+
+    def __init__(
+        self,
+        trajectory_file: Path,
+        topology_file: Path | None = None,
+        xyz_cols: list[str] | None = None,
+    ) -> None:
+        """Initialize the RDF object and import the trajectory data.
+
+        Args:
+            trajectory_file (Path):
+                Path to the trajectory file (e.g. `.xyz` or `.xtc`).
+
+            topology_file (Path | None, optional):
+                Path to the topology file if required (e.g. `.gro`).
+
+            xyz_cols (list[str] | None, optional):
+                List of column names for XYZ format
+                files (e.g., `["Particle Type", "X", "Y", "Z"]`).
+        """
+        # Check number of args and setup the trajectory files
+        if topology_file is None:
+            self.trajectory = [Path(trajectory_file)]
```

Comment:
You do not use this variable elsewhere outside of the init, right?

So you can remove the `self.`